src/lsp/cobol_lsp/lsp_completion.ml

```diff
@@ -97,73 +54,69 @@ let range (pos:Position.t) text =
     let rec inner line_start i : int * int =
       let line_end = try String.index_from s line_start '\n' with _ -> String.length s in
       if i >= n
-        then (line_start, line_end)
-        else inner (line_end+1) (i+1)
-      in
-      let (start,end_) = inner 0 0 in
-      String.sub s start (end_-start)
-      in
-    let text_line = get_nthline (Lsp.Text_document.text text) line in
-    let index = List.fold_left (fun acc delimiter ->
+      then (line_start, line_end)
+      else inner (line_end+1) (i+1)
+    in
+    let (start,end_) = inner 0 0 in
+    String.sub s start (end_-start)
+  in
+  let text_line = get_nthline (Lsp.Text_document.text text) line in
+  let index = List.fold_left (fun acc delimiter ->
       let current =
         try 1 + String.rindex_from text_line (character - 1) delimiter
         with _ -> 0 in
       max acc current) 0 delimiters in
-    let position_start = Position.create ~character:index ~line in
-    Range.create ~start:position_start ~end_:pos
+  let position_start = Position.create ~character:index ~line in
+  Range.create ~start:position_start ~end_:pos
 
 let to_completion_item ~kind ~range qualnames =
   List.flatten @@ List.map (function
-    | Cobol_ptree.Name name -> [~&name]
-    | Qual (name,_) as qualname ->
+      | Cobol_ptree.Name name -> [~&name]
+      | Qual (name,_) as qualname ->
         [~&name; Pretty.to_string "%a" Cobol_ptree.pp_qualname qualname]
-    | _ -> []) qualnames |>
-    List.map @@ completion_item ~kind ~range
+      | _ -> []) qualnames |>
+  List.map @@ completion_item ~kind ~range
+
+let string_of_K tokens =
+  let pp ppf =
+    let rec inner ?(space_before=true) = function
+      | [] -> ()
+      | hd::tl ->
+        let token' = hd &@ Srcloc.dummy in
+        if hd <> Cobol_parser.Tokens.PERIOD && space_before then Fmt.sp ppf ();
+        Cobol_parser.INTERNAL.pp_token ppf token';
+        inner tl
+    in inner ~space_before:false
+  in Pretty.to_string "%a" pp (Basics.NEL.to_list tokens)
```

Comment:
```suggestion
  in
  Pretty.to_string "%a" pp (Basics.NEL.to_list tokens)
```
Maybe that would be clearer with a match on `PERIOD` instead of an equality test on `hd`.

Reply:
Please add a comment "TODO: Cobol_parser.Tokens.pp".